libs/nucs/src/policy.rs

```diff
@@ -74,8 +74,8 @@ pub struct OperatorPolicy {
 }
 
 impl OperatorPolicy {
-    fn evaluate(&self, value: &serde_json::Value) -> bool {
-        let value = self.selector.apply(value);
+    fn evaluate(&self, value: &serde_json::Value, context: &HashMap<&str, serde_json::Value>) -> bool {
```

Comment:
maybe better to have keys as `String`? It works now with hand written tests with 'static, but I am not sure this will work with borrowed keys from temp buffers from CLI json?

Reply:
> I am not sure this will work with borrowed keys from temp buffers from CLI json

That is in [this PR already](https://github.com/NillionNetwork/nilvm/pull/104/files#diff-bb7cc55806bc509e3a0fc3bccef8daec0d3e6628465df90c41d20859b5d1961aR92-R95), note that we're taking `&str` not `&'static str`. This hashmap will be built from concrete keys, e.g. you'll say `{"headers": { ... }}` so I think actually `&str` is more common than constructing `String`s.

e.g. I can see this being actual code:

```rust
let headers = ...;
let context = HashMap::from([
    ("headers", headers),
]);
validator.validate(token, parameters, &context);
```